src/main/java/org/apache/maven/buildcache/checksum/MavenProjectInput.java

```diff
@@ -165,45 +185,101 @@ public MavenProjectInput(
-                filteredOutPaths.add(path);
+                addToExcludedSection(propertyValue, true);
+
                 if (LOGGER.isDebugEnabled()) {
                     LOGGER.debug(
-                            "Adding an excludePath from property '{}', values is '{}', path is '{}' ",
-                            propertyName,
-                            propertyValue,
-                            path);
+                            "Adding an excludePath from property '{}', value is '{}'", propertyName, propertyValue);
                 }
             }
         }
         CacheUtils.debugPrintCollection(
-                LOGGER,
-                filteredOutPaths,
-                "List of excluded paths (checked either by fileName or by startsWith prefix)",
-                "Path entry");
+                LOGGER, inputExcludePathMatcherString, "List of excluded glob patterns", "Pattern");
 
         this.fileComparator = new PathIgnoringCaseComparator();
     }
 
+    private void addDefaultExcludeSection(MavenProject project) {
+
+        Build build = project.getBuild();
+        Path buildDirectoryPath = normalizedPath(build.getDirectory());
+        Path outputDirectoryPath = normalizedPath(build.getOutputDirectory());
+        Path testOutputDirectoryPath = normalizedPath(build.getTestOutputDirectory());
+        addToExcludedSection(
+                convertToPathMatcherFileSeperator(buildDirectoryPath.toString()) + GLOB_SX_FULL_SUB_TREE,
+                false); // target by default
+
+        if (!outputDirectoryPath.startsWith(buildDirectoryPath)) {
+            addToExcludedSection(
+                    convertToPathMatcherFileSeperator(outputDirectoryPath.toString()) + GLOB_SX_FULL_SUB_TREE,
+                    false); // target/classes by default
+        }
+        if (!testOutputDirectoryPath.startsWith(buildDirectoryPath)) {
+            addToExcludedSection(
+                    convertToPathMatcherFileSeperator(testOutputDirectoryPath.toString()) + GLOB_SX_FULL_SUB_TREE,
+                    false); // target/test-classes by default
+        }
+    }
+
+    private String convertToPathMatcherFileSeperator(String path) {
+        return path.replace("\\", "/");
+    }
+
+    /**
+     * Add a value from the excluded section list to the directories and/or the filenames ban list.
+     * @param excludedValue a value from the exclude list
+     */
+    private void addToExcludedSection(String excludedValue, boolean addProjectBaseDir) {
+
+        String pathMatcherGlob = GLOB_PX
+                +
+                // Add the base directory to any input directly coming from user configuration
+                (addProjectBaseDir ? baseDirectoryGlobPrefix : "")
+                +
+                // If the glob start with "/", we remove it since it's already added in the added basedir glob
+                (excludedValue.startsWith("/") ? excludedValue.substring(1) : excludedValue);
```

Comment:
Stripping the leading `/` from user-supplied parameters might be confusing. a common convention is that `/` denotes an absolute path.

Reply:
Got it. What about forbidding a leading "/" ? I don't see any use of an absolute glob exclusion.